### DISCUSSION TITLE
How vscode determine a window restart is needed after an extension is upgraded or installed ?

### DISCUSSION THREAD

**Discussion by jianyexi** [microsoft/vscode-discussions#714] [Extension Development QnA] [answered]
is there any doc related to it. I want to opt into the extension upgrade process, as I need to clean up some files which may be locked by the old version extension process.

**Comment by olguzzar** [comment, marked as answer]
VS Code automatically checks for extension updates, and as long as a user has the `extensions.autoUpdate` setting enabled (which is the default), this behavior will occur. You can read more in [this section](https://code.visualstudio.com/docs/editor/extension-marketplace#_extension-autoupdate) of the docs.

If you are wondering how to update the version of an extension that you are publishing, you can read more about that [here](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#autoincrement-the-extension-version).

**Reply by jianyexi** [reply]
Neither, when you upgrade an extension of vscode sometimes you got 'Reload required' here, but some extensions are not. I wonder why and how to apply hot update to my extension without reloading it.
![image](https://github.com/microsoft/vscode-discussions/assets/59603451/ac244f16-b4cb-4694-b521-5566d9996080)

**Reply by r3m0t** [reply]
If the extension hasn't been activated yet then no reload is required. Specifically themes, language packs, and other extensions that don't have any JS like language grammars.
